### PR TITLE
Add contract-wrapper

### LINKED ISSUE #8: Make L1 adaptor more abstract
## Why

To support another L1

## What

- abstracts "management privatekey"(such as Wallet class of etheres.js)
- generic address type

### PR DESCRIPTION
**Pull request by syuhei176**
Close #8 

* [x] Wallet is abstracted? There is IWallet interface.
* [x] Address is abstracted? The address is string type.
* [x] Is there the interface of EventWatcher
* [x] Is there the interface of CommitmentContract

### CODE REVIEW

**Comment by tkmct** [comment]
oh, we need CI.

**Comment by syuhei176** [comment]
Yes, we need CI first.

**Comment by syuhei176** [comment]
updated

File: src/events/interfaces/IEventWatcher.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+export type EventHandler = (e: any) => void
+export type CompletedHandler = () => void
+export type ErrorHandler = (err: Error) => void
+
+export interface IEventWatcher {
```
Comment (syuhei176) [inline src/events/interfaces/IEventWatcher.ts]: This is Eventwatcher's interface.

File: src/contract/interfaces/ICommitmentContract.ts [hunk 2]
```diff
@@ -0,0 +1,5 @@
+import { Bytes } from '../../types'
+
+export interface ICommitmentContract {
```
Comment (syuhei176) [inline src/contract/interfaces/ICommitmentContract.ts]: This is commitment contrcat's interface.

**Pull request merged**